src/metabase/actions/execution.clj

```diff
@@ -114,37 +114,39 @@
         extra-parameters         (set/difference (set (keys request-parameters))
                                                  (set (keys slug->field-name)))
         pk-field                 (first pk-fields)
-        simple-parameters        (update-keys request-parameters (comp keyword slug->field-name))
+        simple-parameters        (->> (update-keys request-parameters (comp keyword slug->field-name))
```

Comment:
Trying to understand this change. The gist is that we want optional parameters to be honored: the action can work and optional params that aren't supplied work. I would have guessed that the implementation would allow for parameters with null values and the optional values spliced in. I'm surprised to see them stripped from the `simple-parameters` map here. I'm not sure I understand the implications of one versus the other.

Reply:
I want to clarify this. There was a FE bug (fixed in this PR) where no values were sent for fields not changed by the user in **basic** action forms, and that's how they previously work. If you type a character into an optional field and remove it, the FE sends `null`, which is handled differently by the backend, and sometimes leads to errors. This change on the backend makes it work the same way with if the parameter is not send or if `null` is send, which totally makes sense for optional parameters in implicit actions. So in other words - this PR is not changing the execution behavior, it just simplifies/fixes the frontend by sending `null` in all cases, and lets the backend handle that case (in the same way the frontend did before).

Reply:
To Dan's point, this change makes it no longer possible for an action to set a non-null value to null. But I don't see how the backend can be smarter about it: if it receives `{favorite_color: null}` there's no way to know if the user no longer has a favorite color or if it just happens that nothing was selected in the FE's color-picker.

I think the real way around this is to do a more major overhaul and let users configure which columns they care about, populate the form with existing values, etc.

Reply:
(@dpsutton @ranquild )